examples/ur5-ik.cpp

```diff
@@ -90,18 +90,19 @@ int main() {
       allocate_shared_eigen_aligned<QuadraticResidualCostTpl<Scalar>>(fn, w1);
   auto cost2 = allocate_shared_eigen_aligned<QuadraticDistanceCostTpl<Scalar>>(
       space, q0, w2);
-  auto cost = std::make_shared<CostSumTpl<Scalar>>(space->nx(), space->ndx());
+  auto cost = std::make_shared<CostSumTpl<Scalar>>(space.nx(), space.ndx());
   cost->addComponent(cost1);
   cost->addComponent(cost2);
 
   Problem problem(space, cost);
 
   constexpr bool has_joint_lims = true;
   if (has_joint_lims) {
-    auto box_cstr = std::make_shared<BoxConstraintTpl<Scalar>>(
-        model.lowerPositionLimit, model.upperPositionLimit);
+    BoxConstraintTpl<Scalar> box_cstr(model.lowerPositionLimit,
+                                      model.upperPositionLimit);
     ConstraintObjectTpl<Scalar> cstrobj(
-        std::make_shared<ManifoldDifferenceToPoint<Scalar>>(space, q0),
+        std::make_shared<ManifoldDifferenceToPoint<Scalar>>(PolyManifold{space},
```

Comment:
Use of shared pointer here. Do we plan to take care of it in this PR or the next?

Reply:
For cost functions and residuals, we plan to do this in a later PR. For now we are doing this for the components of proxsuite-nlp which are used downstream in aligator.